### ISSUE TITLE
Add HealthCheckCustomConfig/FailureThreshold support to aws_service_discovery_service

### ISSUE THREAD

**Issue by jeffnappi** [terraform-providers/terraform-provider-aws#4082]
When configuring an AWS Service Discovery _Service_ for a private DNS namespace, it is currently required to set the HealthCheckCustomConfig FailureThreshold.

This is documented here: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-discovery.html

The example AWS gives is:
`aws servicediscovery create-private-dns-namespace --name staging --vpc vpc-abcd1234 --region us-east-1` ... 
`aws servicediscovery create-service --name myapplication --dns-config NamespaceId="ns-uejictsjen2i4eeg",DnsRecords=[{Type="A",TTL="300"}] --health-check-custom-config FailureThreshold=1 --region us-east-1`

Although labeled as a complex type, AWS currently only describes one available field for HealthCheckCustomConfig: FailureThreshold.

Here's an example of what a config might look like:

```
resource "aws_service_discovery_service" "my_service" {
  name = "my_service_name"
  
  health_check_custom_config {
    failure_threshold = 1
  }
  
  dns_config {
    dns_records {
      ttl  = 0
      type = "A"
    }

    namespace_id   = "${...namespace_id}"
    routing_policy = "MULTIVALUE"
  }
}
```

**Comment by jeffnappi** [comment]
If you do not set the HealthCheckCustomConfig FailureThreshold your tasks will be killed within about 3 minutes with this error: `Timeout waiting for service discovery instance provisioning to complete.`

**Comment by ryanvm** [comment]
I can verify this bug.

As of commit 82c8a43c1a0c9baf88db73a2b27d883e5a9b1303, it seems to be impossible to use Service Discovery with _private_ Fargate instances because they keep timing out during provisioning. Additionally, I was also able to confirm the fix from @jeffnappi in PR #4083. After building with his PR and adding the requisite stanza to my `aws_service_discovery_service` resources, everything is working. Thanks Jeff!

**Comment by bflad** [comment]
The enhancement has been merged into master and will release with v1.15.0 of the AWS provider, likely mid next week. We should probably also test/document the failure scenario to help future travelers.
